packages/angular_devkit/build_angular/src/builders/karma/application_builder.ts

```diff
@@ -64,13 +66,15 @@ function injectKarmaReporter(
 
     private startWatchingBuild() {
       void (async () => {
-        for await (const buildOutput of buildApplicationInternal(
-          {
-            ...buildOptions,
-            watch: true,
-          },
-          context,
-        )) {
+        // This is effectively "for await of but skip what's already consumed".
+        let isDone = false; // to mark the loop condition as "not constant".
+        while (!isDone) {
+          const { done, value: buildOutput } = await buildIterator.next();
+          if (done) {
+            isDone = true;
+            break;
+          }
```

Comment:
Can this `for..await` the buildIterator here or is the manual iterator usage required?

Reply:
Unfortunately `for await..of` needs an `AsyncIterable` because it wants to create a fresh (and reset) `AsyncIterator` itself. From what I can gather, we can't (easily) use the nicer syntax here.